### PR TITLE
Issue 14513 implements crumbtrails in the new chrome

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/app/api/services/dot-menu.service.ts [hunk 1]
```diff
@@ -78,10 +78,21 @@ export class DotMenuService {
...
+        return this.getDotMenu(portletId)
+            .map((menu: DotMenu) => menu.id);
+    }
+
+    getDotMenu(portletId: string): Observable<DotMenu> {
         return this.loadMenu()
             .flatMap((menus: DotMenu[]) => menus)
-            .find((menu: DotMenu) => menu.menuItems.some((menuItem) => menuItem.id === portletId))
-            .map((menu: DotMenu) => menu.id);
+            .find((menu: DotMenu) => menu.menuItems.some((menuItem) => menuItem.id === portletId));
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-menu.service.ts]: add type to `menuItem`

File: src/app/view/components/_common/dot-crumb-trail/dot-crumb-trail.component.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@Component({
+    selector: 'dot-crumb-trail',
+    templateUrl: './dot-crumb-trail.component.html'
+})
+export class DotCrumbTrailComponent {
+    public crumbs: MenuItem[];
+
+    constructor(private crumbTrailService: CrumbTrailService, private sanitizer: DomSanitizer) {
+        this.crumbTrailService.crumbTrail.subscribe(crumsTrail => {
+            this.crumbs = crumsTrail.crumbs.map((crumb) => {
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-crumb-trail/dot-crumb-trail.component.ts]: add types to crumb

File: src/app/api/services/dot-menu.service.ts [hunk 3]
```diff
@@ -78,10 +78,21 @@ export class DotMenuService {
      * @param portletId MenuItems id
      */
     getDotMenuId(portletId: string): Observable<string> {
+        return this.getDotMenu(portletId)
+            .map((menu: DotMenu) => menu.id);
+    }
+
+    getDotMenu(portletId: string): Observable<DotMenu> {
```
Comment (fmontes) [inline src/app/api/services/dot-menu.service.ts]: doc

File: src/app/api/services/dot-menu.service.ts [hunk 4]
```diff
@@ -78,10 +78,21 @@ export class DotMenuService {
...
         return this.loadMenu()
             .flatMap((menus: DotMenu[]) => menus)
-            .find((menu: DotMenu) => menu.menuItems.some((menuItem) => menuItem.id === portletId))
-            .map((menu: DotMenu) => menu.id);
+            .find((menu: DotMenu) => menu.menuItems.some((menuItem: DotMenuItem) => menuItem.id === portletId));
+    }
+
+    getDotMenuItem(portletId: string): Observable<DotMenuItem> {
+        return this.loadMenu()
+            .flatMap((menus: DotMenu[]) => menus)
```
Comment (fmontes) [inline src/app/api/services/dot-menu.service.ts]: Can't you just do `.flatMap((menus: DotMenu[]) => menus.menuItem)` and have only one `flatMap`?

File: src/app/app-routing.module.ts [hunk 5]
```diff
@@ -78,7 +79,10 @@ const PORTLETS_IFRAME = [
                 children: [
                     {
                         loadChildren: 'app/portlets/dot-porlet-detail/dot-portlet-detail.module#DotPortletDetailModule',
-                        path: ':asset'
+                        path: ':asset',
+                        data: {
+                            excludeFromCrumbTrail: true
```
Comment (fmontes) [inline src/app/app-routing.module.ts]: Why not... `crumbtrail: false` and by default is true?

File: src/app/view/components/_common/dot-crumb-trail/dot-crumb-trail.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+        fixture = DOTTestBed.createComponent(DotCrumbTrailComponent);
+        de = fixture.debugElement;
+    });
+
+    it('should have a p-breadcrumb', () => {
+        expect(de.query(By.css('p-breadcrumb')) !== null).toBe(true);
+    });
+
+    describe('DotCrumbTrailComponent', () => {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-crumb-trail/dot-crumb-trail.component.spec.ts]: Why a second describe here?

File: src/app/view/components/_common/dot-crumb-trail/services/crumb-trails-guard.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+        crumbTrailsGuardService = TestBed.get(CrumbTrailsGuardService);
+        crumbTrailService = TestBed.get(CrumbTrailService);
+        mockRouterStateSnapshot = jasmine.createSpyObj<RouterStateSnapshot>('RouterStateSnapshot', ['toString']);
+        mockActivatedRouteSnapshot = jasmine.createSpyObj<ActivatedRouteSnapshot>('ActivatedRouteSnapshot', [
+            'toString'
+        ]);
+    });
+
+    it('should push then Crumb trails', () => {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-crumb-trail/services/crumb-trails-guard.service.spec.ts]: This description is like: WHATTTT?

File: src/app/view/components/_common/dot-crumb-trail/services/crumb-trails-guard.service.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Route Guard that checks if a User have access to the specified Content Type.
+ */
+@Injectable()
+export class CrumbTrailsGuardService implements CanActivateChild {
+
+    constructor(private crumbTrailService: CrumbTrailService) {}
+
+    canActivateChild(_route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
+        this.crumbTrailService.activatedRoute = _route;
```
Comment (fmontes) [inline src/app/view/components/_common/dot-crumb-trail/services/crumb-trails-guard.service.ts]: Why don't just access the `ActivatedRouteSnapshot` in the `CrumbTrailService`?

File: src/app/view/components/_common/dot-crumb-trail/services/dot-crumb-trail.service.ts [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+                menuLabel: values[1],
+                parentMenuLabel: values[2]
+            })),
+            take(1)
+        );
+    }
+
+    private getMenuLabel(portletName: string): Observable<string> {
+        if (portletName.startsWith('edit-page')) {
+            return this.dotMenuService.getDotMenu('site-browser')
```
Comment (fmontes) [inline src/app/view/components/_common/dot-crumb-trail/services/dot-crumb-trail.service.ts]:
```typescript
return this.dotMenuService.getDotMenu(portletName.startsWith('edit-page') ? 'site-browser' : portletName)
    .map(dotMenu => dotMenu ? dotMenu.name : null);
```

File: src/app/view/components/_common/dot-crumb-trail/services/dot-crumb-trail.service.ts [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+    messageKey: string;
+    dataId?: string;
+    label?: string;
+    url: string;
+    queryParams?: {
+        [k: string]: any;
+    };
+}
+
+interface Messages {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-crumb-trail/services/dot-crumb-trail.service.ts]: `DotCrumbtrailMessages` or something?

File: src/app/view/components/dot-navigation/dot-navigation.component.ts [hunk 11]
```diff
@@ -42,4 +49,8 @@ export class DotNavigationComponent implements OnInit {
     isActive(id: string) {
         return this.dotNavigationService.isActive(id);
     }
+
+    private isMetaKeyPress(event: MouseEvent): boolean {
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/dot-navigation.component.ts]: cool

**Pull request closed without merging**